### ISSUE TITLE
Allow setting logrotate's maxsize via Taupage config

### ISSUE THREAD

**Issue by regispl** [zalando-stups/taupage#254]
At the moment there are 3 logrotate settings exposed via Taupage: `application_logrotate_size`, `application_logrotate_interval`, `application_logrotate_rotate`. What we're missing at the moment is `maxsize`:

```
maxsize
  Log files are rotated when they grow bigger than size bytes even before the 
  additionally specified time interval (daily, weekly, monthly, or yearly). The 
  related size option is similar except that it is mutually exclusive with the time 
  interval options, and it causes log files to be rotated without regard for the 
  last rotation time. When maxsize is used, both the size and timestamp of a log 
  file are considered.
```
Emphasis on: **Log files are rotated when they grow bigger than size bytes even before the additionally specified time interval**

The reason for adding it is that in some failure modes some services (e.g. Kafka when it can't find leader for partition) can spit GBs of WARN / ERROR log messages in few **hours** (exactly 1.2 GB as I'm writing this ticket 😉 ), which may result in running out of disk space, which then may turn an already serious issue into a complete disaster.

We don't think it makes too much sense to overprovision GBs of disk space for logs that - in general - should not grow any larger than low hundreds of MBs per week, only to account for situations like this, so being able to rotate logs based on maxsize would be useful.

Does it make sense to you?

Also, it looks like logrotate is run by default via crontab on daily basis only. It means that the hourly rotation doesn't make too much sense (it does in terms of the granularity of the log archives created, but if logs grow fast enough to overfill the disk in less than a day it won't help anyway)? **Is there a way to configure logrotate cron via Taupage config to be run hourly rather than daily?** We'd need to change it too for the `maxsize` option to make sense in the case as the one I described above.

**Comment by apfeiffer85** [comment]
You can change the application_logrotate_interval = "hourly" and then it will rotate your logs every hour. 

Docs: http://stups.readthedocs.io/en/latest/components/taupage.html#configuration-option-explanation

**Comment by regispl** [comment]
Unfortunately it won't work - setting this property is not changing how often logrotate runs. From `man logrotate` ( http://manpages.ubuntu.com/manpages/xenial/en/man8/logrotate.8.html ):

```
hourly    Log files are rotated every hour. Note that usually logrotate is
              configured  to  be  run  by  cron daily. You have to change this
              configuration and run logrotate hourly  to  be  able  to  really
              rotate logs hourly.
```

From what I can see ( https://github.com/zalando-stups/taupage/blob/31622212d07140a4f532e34b0d51e78679c5571b/runtime/opt/taupage/init.d/86-adjust-application-logrotate ) setting `application_logrotate_interval` is only setting the logrotate property, but logrotate will only run daily anyway, which is not solving the problem with quickly growing logs as they may overfill the disk anyway 😞 

There's event a comment there saying: `hourly is only possible if we also adjust the logrotate cronjob`, but I can't see that cronjob being adjusted anywhere. Am I missing something?

**Comment by apfeiffer85** [comment]
Yes, thats true. And we do change the cronjob to hourly and also other Teams use this for hourly logrotate. So set the interval to hourly and maybe also the application_logrotate_size to 100m or something else then it will definitely rotate your logs hourly. 
If you use the latest taupage AMI .... 

Again the link to the hourly cronjob: 
https://github.com/zalando-stups/taupage/tree/master/runtime/etc/cron.hourly

**Comment by regispl** [comment]
Thanks, when I read your first reply for the first time, the cron link wasn't there and I probably didn't notice that you updated it later.

OK, I'm using whatever AMI Kafka appliance (buku) is using, which might not be the most recent one for Kafka 0.8. I'll check it, thanks :-)

**Comment by bocytko** [comment]
Being able to set `maxsize` would solve the issues described in #176 as it rotates logs even before the hourly cron kicks in.

**Comment by apfeiffer85** [comment]
**No** it only rotates the logs regarding the size and not the `interval` setting but this has nothing to do with the Cronjob itself. So `maxsize` will not attend to the `interval` setting (thats right). So if the log is larger as `maxsize` it will start rotating independent of the `interval` setting. 

But if you would like to rotate more frequently we have to adjust the Cronjob as well. Because otherwise logrotate will not run and check the log files. 

My suggestion is run the Cronjob every 10Min. 

Would this solve the log size problem ? 

Currently we also have the setting `delaycompress` we could also remove this to save more disk space.

**Comment by bocytko** [comment]
> But if you would like to rotate more frequently we have to adjust the Cronjob as well. Because otherwise logrotate will not run and check the log files.

The intended behavior is: rotate logfile either if `maxsize` is exceeded or if the `interval` has passed.
If running the cronjob more often assures this, then let's go for it.
